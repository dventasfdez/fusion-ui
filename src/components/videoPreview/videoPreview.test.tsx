import React from 'react';
import renderer from 'react-test-renderer';
import {render} from '@testing-library/react';
import VideoPreview from './videoPreview';
import VideoPreviewImage from './videoPreviewImage';
import VideoPreviewTitle from './videoPreviewTitle';
import VideoPreviewSubtitle from './videoPreviewSubtitle';
import VideoPreviewTimer from './videoPreviewTimer';
import VideoPreviewTag from './videoPreviewTag';

const videoPreviewExample = (props?: any) => (
  <VideoPreview {...props}>
    <VideoPreviewImage data-testid="video-preview-img">
      <img src="https://th.bing.com/th/id/OIP.sfpxOWBm9ecBIcKQwsG1fwHaE7" />
    </VideoPreviewImage>
    <VideoPreviewTitle data-testid="video-preview-title">This is the title of the video</VideoPreviewTitle>
    <VideoPreviewSubtitle data-testid="video-preview-subtitle">And this is the subtitle of the video</VideoPreviewSubtitle>
    <VideoPreviewTag data-testid="video-preview-tag">TAG</VideoPreviewTag>
    <VideoPreviewTimer data-testid="video-preview-timer">05:55</VideoPreviewTimer>
  </VideoPreview>
);
const videoPreviewExampleOnlyImg = (props?: any) => (
  <VideoPreview {...props}>
    <VideoPreviewImage data-testid="video-preview-img">
      <img src="https://th.bing.com/th/id/OIP.sfpxOWBm9ecBIcKQwsG1fwHaE7" />
    </VideoPreviewImage>
  </VideoPreview>
);

test('Video preview should render', () => {
  const component = renderer.create(videoPreviewExample());
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Video preview small should render', () => {
  const component = renderer.create(videoPreviewExample({large: true}));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Video preview large should render', () => {
  const component = renderer.create(videoPreviewExample({small: true}));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Video preview only with image should render', () => {
  const component = renderer.create(videoPreviewExampleOnlyImg());
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Display video preview with video preview image', () => {
  const {container} = render(videoPreviewExample());
  const videoImage = container.getElementsByClassName('video-preview-image')[0];
  expect(videoImage).toBeInTheDocument();
});

test('Display video preview with title', () => {
  const {container} = render(videoPreviewExample());
  const title = container.getElementsByClassName('video-preview-title')[0];
  expect(title).toBeInTheDocument();
});

test('Display video preview with subtitle', () => {
  const {container} = render(videoPreviewExample({large: true}));
  const subtitle = container.getElementsByClassName('video-preview-subtitle')[0];
  expect(subtitle).toBeInTheDocument();
});

test('Display video preview with timer', () => {
  const {container} = render(videoPreviewExample());
  const timer = container.getElementsByClassName('video-preview-timer')[0];
  expect(timer).toBeInTheDocument();
});
