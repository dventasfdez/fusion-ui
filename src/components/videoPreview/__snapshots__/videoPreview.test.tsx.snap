// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video preview large should render 1`] = `
<div
  className="video-preview_small "
>
  <div
    className="video-preview-body"
  >
    <div
      className="video-preview-image "
      data-testid="video-preview-img"
    >
      <img
        src="https://th.bing.com/th/id/OIP.sfpxOWBm9ecBIcKQwsG1fwHaE7"
      />
    </div>
  </div>
  <div
    className="video-preview-footer"
  >
    <div
      className="video-preview-footer-left"
    >
      <div
        className="video-preview-timer "
        data-testid="video-preview-timer"
      >
        05:55
      </div>
    </div>
    <div
      className="video-preview-footer-right"
    >
      <div
        className="video-preview-play-button-container"
      >
        <div
          className="video-preview-small-play-title"
        >
          play video
        </div>
        <span
          className="material-icons video-preview-small-play-button"
        >
          play_circle_filled
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Video preview only with image should render 1`] = `
<div
  className="video-preview "
>
  <div
    className="video-preview-body"
  >
    <span
      className="material-icons video-preview-play-button video-preview-medium-play-button"
    >
      play_circle_filled
    </span>
    <div
      className="video-preview-image "
      data-testid="video-preview-img"
    >
      <img
        src="https://th.bing.com/th/id/OIP.sfpxOWBm9ecBIcKQwsG1fwHaE7"
      />
    </div>
  </div>
</div>
`;

exports[`Video preview should render 1`] = `
<div
  className="video-preview "
>
  <div
    className="video-preview-body"
  >
    <div
      className="status-tag "
      data-testid="video-preview-tag"
    >
      TAG
    </div>
    <span
      className="material-icons video-preview-play-button video-preview-medium-play-button"
    >
      play_circle_filled
    </span>
    <div
      className="video-preview-image "
      data-testid="video-preview-img"
    >
      <img
        src="https://th.bing.com/th/id/OIP.sfpxOWBm9ecBIcKQwsG1fwHaE7"
      />
    </div>
  </div>
  <div
    className="video-preview-footer"
  >
    <div
      className="video-preview-footer-left"
    >
      <div
        className="video-preview-title "
        data-testid="video-preview-title"
      >
        This is the title of the video
      </div>
    </div>
    <div
      className="video-preview-footer-right"
    >
      <div
        className="video-preview-timer "
        data-testid="video-preview-timer"
      >
        05:55
      </div>
    </div>
  </div>
</div>
`;

exports[`Video preview small should render 1`] = `
<div
  className="video-preview_large "
>
  <div
    className="video-preview-body"
  >
    <div
      className="status-tag "
      data-testid="video-preview-tag"
    >
      TAG
    </div>
    <span
      className="material-icons video-preview-play-button "
    >
      play_circle_filled
    </span>
    <div
      className="video-preview-image "
      data-testid="video-preview-img"
    >
      <img
        src="https://th.bing.com/th/id/OIP.sfpxOWBm9ecBIcKQwsG1fwHaE7"
      />
    </div>
  </div>
  <div
    className="video-preview-footer"
  >
    <div
      className="video-preview-footer-left"
    >
      <div
        className="video-preview-title "
        data-testid="video-preview-title"
      >
        This is the title of the video
      </div>
      <div
        className="video-preview-subtitle "
        data-testid="video-preview-subtitle"
      >
        And this is the subtitle of the video
      </div>
    </div>
    <div
      className="video-preview-footer-right"
    >
      <div
        className="video-preview-timer "
        data-testid="video-preview-timer"
      >
        05:55
      </div>
    </div>
  </div>
</div>
`;
